### ISSUE TITLE
Installation problems in Windows 10: "error C2872: 'std': ambiguous symbol" and DLL warnings

### ISSUE THREAD

**Issue by JFornt** [bamsumit/slayerPytorch#15]
Hi, I have been trying to install Slayer in Windows 10 without much success. When I run 'setup.py install', I get a ton of warning messages plus the error mentioned in #2 

After applying the suggested changes to `argument_spec.h` and `cast.h` (I was not able to find any `module.h` file in the pytorch folder), I get the error:

`C:/Users/renan/Anaconda3/envs/SLAYER_FINAL/lib/site-packages/torch/include/torch/csrc/api/include\torch/nn/pimpl.h(132): error C2872: 'std': ambiguous symbol`

As well as all the warnings (see error log), that look like:

`warning: base class dllexport/dllimport specification differs from that of the derived class`

`warning: field of class type without a DLL interface used in a class with a DLL interface`

`warning: dllexport/dllimport conflict with "caffe2::TypeMeta::_typeMetaDataInstance [with T=caffe2::detail::_Uninitialized]"
(452): here; dllexport assumed`

I am using a virtual environment with Anaconda3, Python 3.7.6 and CUDA version: 9.2. I have installed all the libraries with the version shown at `requirements.txt`.

The NVIDIA CUDA seems to work properly. I am able to run `deviceQuery` and the samples.

As the latest versions of VS2017 are not compatible with CUDA 9.2, I am running it on the Developer Command Prompt for VS2015.

Am I doing something wrong? Thanks for your help.

I attach the (very long) error log as a .txt file:

[ErrorLog.txt](https://github.com/bamsumit/slayerPytorch/files/4240817/ErrorLog.txt)

**Comment by bamsumit** [comment]
Hi, it is common to see tons of warning during compilation. I did not encounter errors like yours. This might be useful: https://discuss.pytorch.org/t/compiler-report-std-ambiguous-symbol-error-under-visual-c-2017/39459 It says " you can fix that manually by prepending `::` before `std` in these headers."

FYI, I tested the windows installation with the following versions:
* cuda 10.1
* torch 1.4.0
* Microsoft (R) C/C++ Optimizing Compiler Version 19.24.28316 for x64
    * just type `cl` in the command line

**Comment by JFornt** [comment]
Thank you for the quick reply.

I tried to fix the `error C2872: 'std': ambiguous symbol` with much success. After prepending `::` it still gave me the same errors.

However, later I tried to install it with torch 1.4.0 and cuda 10.1. This time I was able to apply all the modifications you mentioned in #2, as the file `module.h` changes significantly from torch 1.1.0 to 1.4.0.

And with that it worked! (Even though I got tons of warning messages, as you mentioned). Also, I ran the unittest and got no errors.

Thanks!!
